code/modules/organs/internal/stomach.dm

```diff
@@ -75,7 +75,9 @@
 		var/mob/living/L = food
 		if((species.gluttonous & GLUT_TINY) && (L.mob_size <= MOB_TINY) && !ishuman(food)) // Anything MOB_TINY or smaller
 			return DEVOUR_SLOW
-		else if((species.gluttonous & (GLUT_SMALLER|GLUT_MESSY)) && owner.mob_size > L.mob_size) // Anything we're larger than
+		else if((species.gluttonous & GLUT_SMALLER) && owner.mob_size > L.mob_size) // Anything we're larger than
+			return DEVOUR_SLOW
+		else if(species.gluttonous & GLUT_MESSY) // Anything we're smaller than
 			return DEVOUR_SLOW
```

Comment:
You should just put this as a separate condition on the if statement before

Reply:
The above if handles when you're trying to eat something which is smaller than you. This one handles things that are bigger. I suppose that I could combine the two into one fuckoff-huge if/or statement, but what I've already done would be neater and easier to read, in my opinion.

Reply:
There used to be an additional segment to the line I added which checked whether you were bigger than the target, but I realised that because of where it is in the else-if chain that the line would be superflous since if you get there in the else-if, whatever you're trying to eat is going to be bigger than you.

Reply:
I worked out the if/or statement and it looks like this:
`else if((species.gluttonous & ((GLUT_SMALLER && (owner.mob_size > L.mob_size)) | (GLUT_MESSY)))`

Reply:
Not really:
`else if(species.gluttonous & GLUT_MESSY || (species.gluttonous & GLUT_SMALLER) && owner.mob_size > L.mob_size))`

Reply:
Donezo